include/libdnf5-cli/output/modulelist.hpp

```diff
@@ -0,0 +1,124 @@
+#include <libdnf5/module/module_sack.hpp>
+#include <libsmartcols/libsmartcols.h>
+
+#include <set>
+#include <string>
+
+namespace libdnf5::cli::output {
+
+
+// module list table columns
+enum { COL_MODULE_NAME, COL_MODULE_STREAM, COL_MODULE_PROFILES, COL_MODULE_SUMMARY };
+
+
+const std::string MODULELIST_TABLE_HINT = _("\nHint: [d]efault, [e]nabled, [x]disabled, [i]nstalled");
+
+
+static struct libscols_table * create_modulelist_table() {
+    struct libscols_table * table = scols_new_table();
+    if (libdnf5::cli::tty::is_interactive()) {
+        scols_table_enable_colors(table, 1);
+    }
+    struct libscols_column * cl = scols_table_new_column(table, "Name", 0.15, 0);
+    scols_column_set_cmpfunc(cl, scols_cmpstr_cells, NULL);
+    scols_table_new_column(table, "Stream", 0.15, SCOLS_FL_TRUNC);
+    scols_table_new_column(table, "Profiles", 0.3, SCOLS_FL_WRAP);
+    scols_table_new_column(table, "Summary", 0.4, SCOLS_FL_WRAP);
+    scols_table_enable_maxout(table, 1);
+    return table;
+}
+
+
+static void add_line_into_modulelist_table(
+    struct libscols_table * table,
+    const char * name,
+    const char * stream,
+    const char * profiles,
+    const char * summary) {
+    struct libscols_line * ln = scols_table_new_line(table, NULL);
+    scols_line_set_data(ln, COL_MODULE_NAME, name);
+    scols_line_set_data(ln, COL_MODULE_STREAM, stream);
+    scols_line_set_data(ln, COL_MODULE_PROFILES, profiles);
+    scols_line_set_data(ln, COL_MODULE_SUMMARY, summary);
+}
+
+
+template <class Query>
+void print_modulelist_table(Query & module_list) {
+    // TODO(pkratoch): Sort the table
+    struct libscols_table * table = create_modulelist_table();
+    std::set<std::pair<std::string, std::string>> name_stream_pairs;
+    for (auto & module_item : module_list) {
+        const std::string & name = module_item.get_name();
+        const std::string & stream = module_item.get_stream();
+        if (!name_stream_pairs.contains(make_pair(name, stream))) {
+            // Get stream string (append [e] or [x] if needed)
+            std::string stream_string = stream;
+            const module::ModuleStatus & status = module_item.get_status();
+            if (status == module::ModuleStatus::ENABLED) {
+                stream_string.append(" [e]");
+            } else if (status == module::ModuleStatus::DISABLED) {
+                stream_string.append(" [x]");
+            } else if (module_item.is_default()) {
+                stream_string.append(" [d]");
+            }
+
+            // Get profile strings (append [d] or [i] if needed)
+            std::vector<std::string> profile_strings;
+            for (auto profile : module_item.get_profiles()) {
+                // TODO(pkratoch): Also show "[i]" for installed profiles
+                // TODO(pkratoch): Make sure defaults are loaded
+                profile_strings.push_back(profile.is_default() ? profile.get_name() + " [d]" : profile.get_name());
```

Comment:
Currently, the default profiles are not marked in the table, but I guess this is connected with the TODO comment above.

Reply:
Yes, the TODO is there for this reason. I just found out that the defaults are loaded, they are just not accessed the correct way. I plan to fix that.